buildsystem/Doxygen.cmake

```diff
@@ -0,0 +1,66 @@
+# Copyright 2014-2015 the openage authors. See copying.md for legal info.
+
+option(GENERATE_DOCS "Generate documentation from the source files." ON)
+
+set(DOXYFILE_TEMPLATE "${BUILDSYSTEM_DIR}/templates/Doxyfile.in")
+set(DOCS_ROOT_DIRECTORY "${CMAKE_BINARY_DIR}/docs")
+
+function(add_doxygen_target TARGET TARGET_SRC_DIR)
+	if(NOT DOXYGEN_FOUND)
+		return()
+	endif()
+
+	set(DOXYFILE "${CMAKE_BINARY_DIR}/${TARGET}_Doxyfile")
+	set(OUTPUT_DIRECTORY "${DOCS_ROOT_DIRECTORY}/${TARGET}")
+	foreach(SRC_DIR ${TARGET_SRC_DIR} ${ARGN})
+		if(NOT IS_ABSOLUTE SRC_DIR)
+			set(SRC_DIR "${CMAKE_CURRENT_SOURCE_DIR}/${SRC_DIR}")
+		endif()
+		set(DOXYGEN_SCAN_FOLDERS "${DOXYGEN_SCAN_FOLDERS} ${SRC_DIR}")
+	endforeach()
+
+	configure_file(${DOXYFILE_TEMPLATE} ${DOXYFILE} @ONLY)
+
+	add_custom_target(${TARGET}_docs
+		COMMAND ${CMAKE_COMMAND} -E make_directory ${OUTPUT_DIRECTORY}
+		COMMAND ${DOXYGEN_EXECUTABLE} ${DOXYFILE}
+		WORKING_DIRECTORY ${CMAKE_BINARY_DIR}
+		COMMENT "Generating docs for ${TARGET}"
+		VERBATIM
+	)
+	add_dependencies(doc ${TARGET}_docs)
+endfunction()
+
+# If disabled, don't even bother going forward
+if(NOT GENERATE_DOCS)
+	add_custom_target(doc
+		COMMAND echo "openage was configured with GENERATE_DOCS option set to OFF."
+		COMMAND echo "Reconfigure the project with this option set as ON to be able to generate docs."
```

Comment:
the `echo` is not wrong, but may lead to problems and difficulties in porting. Is there any `cmake -E touch` like variant for printing messages?